smart_contracts/contract/src/contract_api/runtime.rs

```diff
@@ -194,37 +194,6 @@ pub fn get_named_arg<T: FromBytes>(name: &str) -> T {
     bytesrepr::deserialize(arg_bytes).unwrap_or_revert_with(ApiError::InvalidArgument)
 }
 
-/// Returns given named argument passed to the host for the current module invocation.
-/// If the argument is not found, returns `None`.
-///
-/// Note that this is only relevant to contracts stored on-chain since a contract deployed directly
-/// is not invoked with any arguments.
-pub fn try_get_named_arg<T: FromBytes>(name: &str) -> Option<T> {
-    let arg_size = get_named_arg_size(name)?;
-    let arg_bytes = if arg_size > 0 {
-        let res = {
-            let data_non_null_ptr = contract_api::alloc_bytes(arg_size);
-            let ret = unsafe {
-                ext_ffi::casper_get_named_arg(
-                    name.as_bytes().as_ptr(),
-                    name.len(),
-                    data_non_null_ptr.as_ptr(),
-                    arg_size,
-                )
-            };
-            let data =
-                unsafe { Vec::from_raw_parts(data_non_null_ptr.as_ptr(), arg_size, arg_size) };
-            api_error::result_from(ret).map(|_| data)
-        };
-        // Assumed to be safe as `get_named_arg_size` checks the argument already
-        res.unwrap_or_revert()
-    } else {
-        // Avoids allocation with 0 bytes and a call to get_named_arg
-        Vec::new()
-    };
-    bytesrepr::deserialize(arg_bytes).unwrap_or_revert_with(ApiError::InvalidArgument)
```

Comment:
Change line 225 to `bytesrepr::deserialize(arg_bytes).ok()`

Reply:
that or .map_err(ApiError::InvalidArgument) I believe, I need to test

Reply:
Works well indeed with ok() and could test it thanks.

Reply:
Function return Result<T, ApiError> now to get any deser issue thanks

Reply:
@darthsiroftardis Changed back to Option https://github.com/casper-network/casper-node/pull/4786/commits/94e878572c6b7d0f4714f20cc86ab3e0734d37b7